crypto/params.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2019, Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/params.h>
+
+static const OSSL_PARAM *ossl_param_locate(const OSSL_PARAM *p, const char *key)
+{
+    if (p != NULL && key != NULL)
+        for (; p->data_type != 0; p++)
+            if (strcmp(key, p->key) == 0)
```

Comment:
Should this be case insensitive?

Reply:
A good question.  The OSSL_PARAMS structures haven't had this specified yet.  Given they are for the core and providers to communication rather than generally application, I'm happy with case sensitive.

Reply:
Actually, they *are* for application<-> communication

Reply:
Okay, I retract my earlier statement.

Reply:
About case sensitivity, I've treated them as sensitive so far, and I can't see that we've specified anything else.

Reply:
I'm happy with a _no_ answer here.

Reply:
No :wink: